src/wallet/wallet2.cpp

```diff
@@ -2262,15 +2263,7 @@ bool wallet2::check_connection(uint32_t *version, uint32_t timeout)
 
   if(!m_http_client.is_connected())
   {
-    net_utils::http::url_content u;
-    net_utils::parse_url(m_daemon_address, u);
-
-    if(!u.port)
-    {
-      u.port = m_testnet ? config::testnet::RPC_DEFAULT_PORT : config::RPC_DEFAULT_PORT;
-    }
-
-    if (!m_http_client.connect(u.host, std::to_string(u.port), timeout))
```

Comment:
Yes, this was intentional but I am wondering whether this should be moved to `wallet2::init` instead. The behavior previously was inconsistent - if someone created `wallet2` and _happened_ to call this function, then it would set the default port. I quick glance at the GUI indicates that by default the port is explicitly listed, but if a user were to change the host without a port, the default is likely added because the GUI calls this function often.